tts-voicerss/src/main/java/org/mobicents/media/server/tts/voicerss/VoiceRssEngine.java

```diff
@@ -0,0 +1,193 @@
+import org.mobicents.media.server.tts.core.TTSEngine;
+
+
+/**
+ * VoiceRssEngine contains all the logic to parse the configuration file and processing text data
+ * 
+ * @author Thanh Tran
+ *
+ */
+public class VoiceRssEngine implements TTSEngine{
+	private final static Logger logger = Logger.getLogger(VoiceRssEngine.class);
+	private final static int BUFFER = 1024 * 8;
+	private static final List<NameValuePair> parameters;
+	static {
+		parameters = new ArrayList<NameValuePair>();
+		parameters.add(new BasicNameValuePair("c", "WAV"));
+		parameters.add(new BasicNameValuePair("f", "8khz_16bit_mono"));
+	}
+	
+	// Define elements of configuration xml file
+	private final static String 	SERVICE_ROOT = "service-root";
+	private final static String 	API_KEY = "apikey";
+	
+	// Configuration xml elements data
+	private String 					myAPIKey;
+	
+	private final URI myURIService;
+	private final Map<String, String> myMapLanguage;
+	
+	public VoiceRssEngine(final String configurationPath) throws ConfigurationException {
+		if ( logger.isInfoEnabled() ) {
+			logger.info("Configuration file: " + configurationPath);
+		}
+		
+		myMapLanguage = new HashMap<String, String>();
+		
+		final URL url = getClass().getResource(configurationPath);
+		final XMLConfiguration configuration = new XMLConfiguration(url);
+
+		myAPIKey = configuration.getString(API_KEY);
+		BasicNameValuePair apiNameValuePair = new BasicNameValuePair("key", myAPIKey);
+		if ( !parameters.contains(apiNameValuePair) ) {
+			parameters.add(apiNameValuePair);
+		} else {
+			// Do nothing
+		}
+		
+		// Initialize the speech synthesizer state.
+		myURIService = URI.create(configuration.getString(SERVICE_ROOT));
+		load(configuration);
+	}
+	
+	@Override
+	public byte[] speak(String text) {
+		byte[] speakData = new byte[BUFFER];
+		List<NameValuePair> query = new ArrayList<NameValuePair>();
+		query.addAll(parameters);
+		//query.add(new BasicNameValuePair("hl", getLanguage(language)));
+		query.add(new BasicNameValuePair("src", text));
+		
+		final HttpPost post = new HttpPost(myURIService);
+		try {
+			UrlEncodedFormEntity entity = new UrlEncodedFormEntity(query, "UTF-8");
+			post.setEntity(entity);
+			final HttpClient client = new DefaultHttpClient();
+			HttpResponse response = client.execute(post);
+			final StatusLine line = response.getStatusLine();
+
+			if ( HttpStatus.SC_OK == line.getStatusCode() ) {
+				Header[] contentType = response.getHeaders("Content-Type");
+				
+				if ( contentType[0].getValue().startsWith("text") ) {
+					final StringBuilder buffer = new StringBuilder();
+					String error = EntityUtils.toString(response.getEntity());
+					logger.error("VoiceRssEngine error: " + error);
+					buffer.append(error);
+					throw new SpeechSynthesizerException(buffer.toString());
+				}
+				
+				if ( logger.isInfoEnabled() ) {
+					logger.info("VoiceRssEngine success!");
+				}
+				
+				InputStream is = response.getEntity().getContent();
```

Comment:
I know it's in a try-catch. The point is, you risk that the stream won't be close if an error happens while reading.
Please check the link I sent you.